examples/systemcalls/systemcalls.c

```diff
@@ -93,7 +121,56 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   
 */
 
-    va_end(args);
+	
+	int status;
+	pid_t pid;
+	int fd = open(outputfile, O_WRONLY|O_TRUNC|O_CREAT, 0644);
+	
+	if (fd < 0) 
+	{ 
+		perror("open"); 
+		abort(); 
+	}
+	
+	pid = fork();
+	
+	if (pid == -1)
+	{
+    	va_end(args);
+		return false;
+	}
+		
+	else if (pid == 0) 
+	{
+		if (dup2(fd, STDOUT_FILENO) < 0) 
+		{ 
+			perror("dup2"); 
+			return false; 
+		}
+    	close(fd);
+    	va_end(args);
+		
+		execv(command[0], command);
+		exit (-1);
+	}
+	
+	else if (pid > 0)
+	{
+		close(fd);
+    	va_end(args);
```

Comment:
@MuktaDarekar look for tabs/spaces differences here.
Also consider whether you could combine logic with the function above.